explorer/executor/executor.go

```diff
@@ -90,6 +90,14 @@ func ExecuteFilterQueries(runtime llx.Runtime, queries []*explorer.Mquery, timeo
 }
 
 func (e *instance) runCode(queries map[string]*explorer.ExecutionQuery, timeout time.Duration) error {
+	if len(queries) == 0 {
+		e.progressReporter.Completed()
+		go func() {
+			e.done <- struct{}{}
```

Comment:
we should circle back around to this. `e.done` should just be closed to signal done. there should be no need to launch a goroutine

Reply:
+1, my approach here should be replaced